app/javascript/controllers/notification_controller.js

```diff
@@ -0,0 +1,90 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="notification"
+export default class extends Controller {
+  static targets = ["notification", "newHappyThingform", "happyThings"]
+
+  connect() {
+    console.log("👻 buh from notifications");
+    console.log("newHappyThingform", this.newHappyThingformTarget);
+    console.log("happyThings", this.happyThingsTarget);
+    console.log("notification", this.notificationTarget);
+
+    this.registration = null;
+
+    navigator.serviceWorker.getRegistration().then(reg => {
+      this.registration = reg;
+      this.checkServiceWorkerRegistration();
+      // Listen for Turbo Stream updates for new happy things
+      this.element.addEventListener("turbo:after-stream-render", () => {
+        const message = "A new happy thing was added!";
+        this.sendNotification(message);
+      });
+    });
+  }
```

Comment:
Will need to read more about what is going on here:
- [Service Worker Api](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API)
- [How to Add Web Push Notifications to your PWA on iOS and Android (The Easy Way)](https://pushalert.co/blog/pwa-push-notifications-ios-android/)

TODO:
- [ ] Get rid of eventListener

Reply:
Will let you do the digging then and come back when this is ready.